### DISCUSSION TITLE
Help needed for pytest_runtest_makereport hook

### DISCUSSION THREAD

**Discussion by harmin-parra** [pytest-dev/pytest-html#716] [Q&A]
Hello

I have a situation writting a hook for `pytest_runtest_makereport` and wanted some advice to improve my code.

I am developing a plugin for pytest-html

It requires a few options and the usage of a fixture for all tests wanting to use the plugin.

Example of option:

```
group = parser.getgroup("my-plugin", "my-plugin")
group.addoption(
    "--browser",
    action="store",
    default=None,
    help="Accepted values: firefox, chrome, chromium, edge or safari",
    choices=("firefox", "chrome", "chromium", "edge", "safari"),
)

@pytest.fixture(scope='session')
def browser(request):
    return = request.config.getoption("--browser")
```

The required fixture:

```
@pytest.fixture(scope='function')
def webdriver(browser):
    // initialize and return a driver for the browser
```

Tests that want to use the plugin need to use the `webdriver` fixture

```
def test_sample(webdriver):
    // The test code
```

Then I wrote a hook for `pytest_runtest_makereport` to modify the HTML report.

In order to do that, I need to access the fixtures used by each test.

Initially I did it with this code:

```
feature_request = item.funcargs['request']
driver = feature_request.getfixturevalue('webdriver')
```

It works well if all tests use the fixture. :grinning:

But if I include a test without fixture that doesn't need the plugin, an exception is thrown in `pytest_runtest_makereport `and no report is produced. :confused:

I "fixed it" with this code:

```
try:
    feature_request = item.funcargs['request']
except Exception as e:
   // The test isn't using any fixture, so return.
    return
driver = feature_request.getfixturevalue('webdriver')
```

My intuition tells me that this fix can be done in a more elegant and pythonic way.

Could you please give me some tips on how to improve the code ?

file: https://github.com/harmin-parra/pytest-selenium-screenshot/blob/main/src/pytest_selenium_screenshot/plugin.py
line : 358

Thank you

**Comment by BeyondEvil** [comment]
Looks like you're trying to re-invent https://github.com/pytest-dev/pytest-selenium (also maintained by me).

**Reply by BeyondEvil** [reply]
pytest-html is a dependency of pytest-selenium FYI

**Comment by harmin-parra** [comment]
out of curiosity, why is `[pytest11] `the entry point for pytest-html plugins ?

**Reply by RonnyPfannschmidt** [reply]
That's legacy from pytest 1.1 introducing them
